nixos/modules/services/networking/dhcpcd.nix

```diff
@@ -208,7 +210,7 @@ in
         wants = [ "network.target" ];
         before = [ "network-online.target" ];
 
-        restartTriggers = [ exitHook ];
+        restartTriggers = optional (config.services.ntp.enable || config.services.openntpd.enable || config.services.chrony.enable || cfg.runHook != "") [ exitHook ];
```

Comment:
I would just add it unconditionally, at worse it will never trigger a restart.

Reply:
In this case an empty file will be executed

Reply:
This `optional` should go on `environment.etc."dhcpcd.exit-hook"` (and be a `mkIf`) instead.

Reply:
Updated.

Reply:
> This `optional` should go on `environment.etc."dhcpcd.exit-hook"` (and be a `mkIf`) instead.

Added  `environment.etc."dhcpcd.exit-hook" = mkIf ...`, but still an empty file is executed in `restartTriggers`. Returned `restartTriggers = optional ...`.

Reply:
That's not what `restartTriggers` does.

Reply:
Updated.